packages/commonwealth/server/workers/sitemap/createSitemapGenerator.spec.ts

```diff
@@ -0,0 +1,38 @@
+import { models } from '@hicommonwealth/model';
+import { expect } from 'chai';
+import { createAsyncWriterMock } from './createAsyncWriter';
+import { createDatabasePaginatorMock } from './createDatabasePaginator';
+import { createSitemapGenerator } from './createSitemapGenerator';
+
+describe('createSitemapGenerator', function () {
```

Comment:
Our test placement pattern is not to put tests alongside the implementation. While we can discuss this and possibly change the pattern in the future, for now, tests should reside in `/test` and not inside `/server**`.

Reply:
I'd really like to change this and put the tests along side the files.  I think this is going to slow down test creation.  Is there a bug/problem that I'm not seeing here?

Reply:
No bug. How would keeping tests in `/test` slow down test creation?

Reply:
IntelliJ has support for refactoring and auto test creation. If you have foo.ts and you "create a test" around foo it will put it next to foo.test.ts.  It will also detect if you rename foo.ts to bar.ts and rename the test. I'm pretty sure they don't detect it if it's not in the same directory so now you added tech debt when you have a foo.spec.ts test that tests bar.ts.  

Java does it closer to what you want (with IntelliJ) where if there is a 'src' dir there is also a 'test' dir but it's more of a convention there because of the way Java creates .jar files as artifacts.  

The other thing is it allows you to find the test quickly and run it so it supports TDD better. You can do this with Java since it knows the mapping between the test and the code and you can just "run test" for code or jump to the tests easily.  

I like to do TDD dev personally and even if the team doesn't really adopt it I'd still like to do it.

Unfortunately though our mocha won't work with my individual tests unless I run it from the command line. I'm still not sure if that's a bug in our environment or my IDE.

Reply:
Ok - we discussed this pattern in office hours and we have previously discussed this with the team as well. For now, I reiterate that we should maintain the current pattern (this file should live in `/test`). We can consider this new pattern but not without an ADR and discussion with the team.

We chose not to place tests alongside code because we are creating lifecycle tests which don't strictly match 1 to 1 with files. Rather lifecycle tests encompass functionality surrounding a specific aggregate/domain and are not limited to a single command/query/policy. This decision is driven by our DDD approach. We also chose not to place tests in `/src` to avoid bloat. @Rotorsoft can explain the rational behind the decision better than I can.

Reply:
@burtonator let me know if you want to sync with the team to discuss unit testing, including existing and pros/cons of new conventions... we are open to improvements in all areas 👍

Reply:
Thanks for the clarification. I'll just go with the existing convention for now. One test won't be an issue and then we can figure out something long term!  👍👍👍